src/profile-logic/merge-compare.js

```diff
@@ -265,6 +265,12 @@ export function mergeProfilesForDiffing(
     );
   }
 
+  // In merged profiles, we don't want to hide any threads: either they've been
+  // explicitely selected by the user, or it's the diffing track.
+  resultProfile.meta.initialVisibleThreads = resultProfile.threads.map(
```

Comment:
Oh, this `initialVisibleThreads` looks like a new addition, interesting. It looks slightly hacky to me to do it inside the meta for the compare view, but probably changing the visible thread code will make it a lot more complicated.